packages/pigeon/lib/generator_tools.dart

```diff
@@ -287,19 +287,51 @@ Iterable<String> _getReferencedTypes(TypeDeclaration type) sync* {
   yield type.baseName;
 }
 
-/// Given an [Api], return the enumerated classes that must exist in the codec
-/// where the enumeration should be the key used in the buffer.
-Iterable<EnumeratedClass> getCodecClasses(Api api) sync* {
-  final Set<String> names = <String>{};
-  for (final Method method in api.methods) {
-    names.addAll(_getReferencedTypes(method.returnType));
-    for (final NamedType argument in method.arguments) {
-      names.addAll(_getReferencedTypes(argument.type));
+/// Recurses into a list of [Api]s and produces a list of all referenced types.
+Set<String> getReferencedTypes(List<Api> apis, List<Class> classes) {
```

Comment:
Why is this returning strings rather than actual type objects?

Reply:
I only ever needed the basename, switched to returning TypeDeclaration